packages/ERTP/src/issuer.js

```diff
@@ -0,0 +1,240 @@
+    assert(
+      purses.length === 0 || purses.length === 1,
+      'purses length must be 0 or 1',
+    );
+
+    // There may be zero, one, or many payments as input to
+    // reallocate. We want to protect against someone passing in
+    // what appears to be multiple payments that turn out to actually
+    // be the same payment (an aliasing issue). The `combine` method
+    // legitimately needs to take in multiple payments, but we don't
+    // need to pay the costs of protecting against aliasing for the
+    // other uses.
+
+    if (payments.length > 1) {
+      const paymentSet = new Set();
+      payments.forEach(payment => {
+        if (paymentSet.has(payment)) {
+          throw new Error('same payment seen twice');
+        }
+        paymentSet.add(payment);
+      });
+    }
+
+    const totalPursesB = purses.map(purseLedger.get).reduce(add, empty);
+    const totalPaymentsB = payments.map(paymentLedger.get).reduce(add, empty);
+    const total = amountMath.add(totalPursesB, totalPaymentsB);
+
+    const newPaymentsTotal = newPaymentBalances.reduce(add, empty);
+    const newPursesTotal = newPurseBalances.reduce(add, empty);
+    const newTotal = amountMath.add(newPaymentsTotal, newPursesTotal);
+
+    // Invariant check
+    assert(amountMath.isEqual(total, newTotal), 'rights were not conserved');
+
+    // commit point
+    payments.map(payment => paymentLedger.delete(payment));
+    purses.map((purse, i) => purseLedger.set(purse, newPurseBalances[i]));
+
+    const newPayments = newPaymentBalances.map(balance => {
+      const newPayment = makePayment();
+      paymentLedger.init(newPayment, balance);
+      return newPayment;
+    });
+
+    return harden(newPayments);
+  };
+
+  const issuer = harden({
+    getBrand: () => brand,
+    allegedName: () => allegedName,
+    getAmountMath: () => amountMath,
+    getMathHelpersName: () => mathHelpersName,
+    makeEmptyPurse: () => {
+      const purse = makePurse();
+      purseLedger.init(purse, amountMath.getEmpty());
+      return purse;
+    },
+    isLive: paymentLedger.has,
+    getBalance: paymentLedger.get,
+    burn: (paymentP, optAmount = undefined) => {
+      const payment = E.unwrap(paymentP);
+      const paymentBalance = paymentLedger.get(payment);
+      assertAmountEqual(paymentBalance, optAmount);
+      paymentLedger.delete(payment);
+      return paymentBalance;
+    },
+    claim: (paymentP, optAmount) => {
+      const srcPayment = E.unwrap(paymentP);
+      const srcPaymentBalance = paymentLedger.get(srcPayment);
+      assertAmountEqual(srcPaymentBalance, optAmount);
+      // Commit point.
+      const [payment] = reallocate(
+        harden({
+          payments: [srcPayment],
+          newPaymentBalances: [srcPaymentBalance],
+        }),
+      );
+      return payment;
+    },
+    combine: (fromPaymentsPArray, totalAmount) => {
+      const fromPaymentsArray = fromPaymentsPArray.map(E.unwrap);
+      const totalPaymentsB = fromPaymentsArray
```

Comment:
What is the `B` suffix about here?

Reply:
I think I was experimenting with reducing `Balance` to 'B' but since there is `A` and `B` elsewhere, it doesn't make much sense. I'll use the full `Balance`